lock/src/main/java/org/infinispan/lock/impl/functions/UnlockFunction.java

```diff
@@ -12,52 +12,74 @@
 import org.infinispan.commons.marshall.MarshallUtil;
 import org.infinispan.functional.EntryView;
 import org.infinispan.lock.impl.entries.ClusteredLockKey;
+import org.infinispan.lock.impl.entries.ClusteredLockState;
 import org.infinispan.lock.impl.entries.ClusteredLockValue;
 import org.infinispan.lock.impl.externalizers.ExternalizerIds;
 import org.infinispan.lock.impl.log.Log;
 
 /**
  * Function that allows to unlock the lock, if it's not already released.
- *
+ * <p>
  * <p>
  * <ul>
  *    <li>If the requestor is not the owner, the lock won't be released. </li>
  *    <li>If the requestId is null, this value does not affect the unlock </li>
- *    <li>If the requestId is not null, the lock will be released only if the requestId and the owner match </li>
+ *    <li>If the requestId is not null, the lock will be released only if the requestId and the owner match</li>
+ *    <li>If lock is already released, nothing happens</li>
  * </ul>
  * <p>
  *
  * @author Katia Aresti, karesti@redhat.com
  * @since 9.2
  */
-public class UnlockFunction implements Function<EntryView.ReadWriteEntryView<ClusteredLockKey, ClusteredLockValue>, Void> {
+public class UnlockFunction implements Function<EntryView.ReadWriteEntryView<ClusteredLockKey, ClusteredLockValue>, Boolean> {
 
    private static final Log log = LogFactory.getLog(UnlockFunction.class, Log.class);
 
    public static final AdvancedExternalizer<UnlockFunction> EXTERNALIZER = new Externalizer();
 
    private final String requestId;
-   private final Object requestor;
+   private final Set<Object> requestors;
 
    public UnlockFunction(Object requestor) {
       this.requestId = null;
-      this.requestor = requestor;
+      this.requestors = Collections.singleton(requestor);
    }
 
-   public UnlockFunction(String requestId, Object requestor) {
+   public UnlockFunction(String requestId, Set<Object> requestors) {
       this.requestId = requestId;
-      this.requestor = requestor;
+      this.requestors = requestors;
    }
 
    @Override
-   public Void apply(EntryView.ReadWriteEntryView<ClusteredLockKey, ClusteredLockValue> entryView) {
+   public Boolean apply(EntryView.ReadWriteEntryView<ClusteredLockKey, ClusteredLockValue> entryView) {
+      log.tracef("unlock request by reqId [%s] requestors %s", requestId, requestors);
```

Comment:
Might as well add the `static final boolean isTraceEnabled` so we don't always invoke these when tracing is disabled.

Reply:
Should do that across all the classes.

Reply:
I have a Jira to add lock names everywhere. I can handle that there, wtyt ?
ISPN-8769

Reply:
sure

Reply:
+1, but it should be `trace` instead of `isTraceEnabled`.

Reply:
👍